src/WebJobs.Extensions.Http/ClaimsPrincipalHelper.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.WebJobs.Extensions.Http
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Net.Http;
+    using System.Runtime.Serialization;
+    using System.Runtime.Serialization.Json;
+    using System.Security.Claims;
+
+    internal static class ClaimsPrincipalHelper
+    {
+        // Shared serializer instance. This is safe for multi-threaded use.
+        private static readonly DataContractJsonSerializer Serializer = GetCustomSerializer();
+
+        public static ClaimsPrincipal FromBindingData(IReadOnlyDictionary<string, object> bindingData)
+        {
+            var req = bindingData.Values.FirstOrDefault(val => val.GetType() == typeof(HttpRequestMessage)) as HttpRequestMessage;
+            if (req == null)
+            {
+                return new ClaimsPrincipal();
+            }
+
+            var claimsPrincipalHeaderValue = GetClaimsPrincipalHeaderValue(req);
```

Comment:
>var [](start = 12, length = 3)

Please use variable type names explicitly when saving the result of a method call (in this case string).